src/Microsoft.AspNetCore.Mvc.Core/Internal/AttributeRoute.cs

```diff
@@ -246,11 +228,66 @@ private class RouteInfo
             
             public int Order { get; set; }
 
-            public string RouteGroup { get; set; }
-
             public string RouteName { get; set; }
 
             public RouteTemplate RouteTemplate { get; set; }
         }
+
+        private class RouteInfoEqualityComparer : IEqualityComparer<RouteInfo>
+        {
+            public static readonly RouteInfoEqualityComparer Instance = new RouteInfoEqualityComparer();
+
+            public bool Equals(RouteInfo x, RouteInfo y)
+            {
+                if (x == null && y == null)
```

Comment:
It doesn't look like we expect `RouteInfo` to be null elsewhere in this class. Do we need the null checks here?